dataplane/cmd/redbeams.go

```diff
@@ -34,49 +34,49 @@ func init() {
+						Usage: "describe a database server",
+						Before: func(c *cli.Context) error {
+							err := cf.CheckConfigAndCommandFlagsWithoutWorkspace(c)
+							if err != nil {
+								return err
+							}
+							return cf.CheckResourceAddressingFlags(c)
+						},
+						Flags:  fl.NewFlagBuilder().AddResourceAddressingFlags().AddAuthenticationFlags().AddOutputFlag().Build(),
 						Action: redbeams.GetDatabaseServer,
 						BashComplete: func(c *cli.Context) {
-							for _, f := range fl.NewFlagBuilder().AddFlags(fl.FlCrn).AddAuthenticationFlags().AddOutputFlag().Build() {
+							for _, f := range fl.NewFlagBuilder().AddResourceAddressingFlags().AddAuthenticationFlags().AddOutputFlag().Build() {
 								fl.PrintFlagCompletion(f)
 							}
 						},
 					},
 					{
-						Name:   "describe-by-name",
-						Usage:  "describe a database server, getting it by name",
-						Before: cf.CheckConfigAndCommandFlagsWithoutWorkspace,
-						Flags:  fl.NewFlagBuilder().AddFlags(fl.FlEnvironmentCrn, fl.FlName).AddAuthenticationFlags().AddOutputFlag().Build(),
-						Action: redbeams.GetDatabaseServerByName,
-						BashComplete: func(c *cli.Context) {
-							for _, f := range fl.NewFlagBuilder().AddFlags(fl.FlEnvironmentCrn, fl.FlName).AddAuthenticationFlags().AddOutputFlag().Build() {
-								fl.PrintFlagCompletion(f)
+						Name:  "get-status",
+						Usage: "get the status of a database server (stack)",
+						Before: func(c *cli.Context) error {
+							err := cf.CheckConfigAndCommandFlagsWithoutWorkspace(c)
+							if err != nil {
+								return err
 							}
+							return cf.CheckResourceAddressingFlags(c)
 						},
-					},
-					{
-						Name:   "get-status",
-						Usage:  "get the status of a database server by CRN",
-						Before: cf.CheckConfigAndCommandFlagsWithoutWorkspace,
-						Flags:  fl.NewFlagBuilder().AddFlags(fl.FlCrn).AddAuthenticationFlags().AddOutputFlag().Build(),
+						Flags:  fl.NewFlagBuilder().AddResourceAddressingFlags().AddAuthenticationFlags().AddOutputFlag().Build(),
 						Action: redbeams.GetDatabaseServerStatus,
 						BashComplete: func(c *cli.Context) {
-							for _, f := range fl.NewFlagBuilder().AddFlags(fl.FlCrn).AddAuthenticationFlags().AddOutputFlag().Build() {
+							for _, f := range fl.NewFlagBuilder().AddResourceAddressingFlags().AddAuthenticationFlags().AddOutputFlag().Build() {
 								fl.PrintFlagCompletion(f)
 							}
 						},
 					},
 					{
-						Name:   "get-status-by-name",
-						Usage:  "get the status of a database server by name",
+						Name:   "create",
+						Usage:  "create a managed database server",
 						Before: cf.CheckConfigAndCommandFlagsWithoutWorkspace,
-						Flags:  fl.NewFlagBuilder().AddFlags(fl.FlEnvironmentCrn, fl.FlName).AddAuthenticationFlags().AddOutputFlag().Build(),
-						Action: redbeams.GetDatabaseServerStatusByName,
+						Flags:  fl.NewFlagBuilder().AddFlags(fl.FlDatabaseServerCreationFile).AddAuthenticationFlags().AddOutputFlag().Build(),
```

Comment:
The usual way how from file is handled:
- submenu below create with name from-file
then, in submenu one mandatory option: --file (but that is more variable)
Examples:
env, credential, blueprint create

Reply:
I've seen those. The reason I didn't follow that pattern is because there isn't another option beyond "from-file", so what's the point of having a subcommand? Just have a file option at the top level, and be done with it.